habitat-lab/habitat/datasets/rearrange/navmesh_utils.py

```diff
@@ -0,0 +1,441 @@
+        # Construct the target trans
+        target_trans = mn.Matrix4.from_(
+            target_rigid_state.rotation.to_matrix(),
+            target_rigid_state.translation,
+        )
+
+        return target_trans
+
+
+def is_navigable_given_robot_navmesh(
+    sim,
+    start_pos,
+    goal_pos,
+    navmesh_offset=([0, 0], [0, 0.15], [0, -0.15]),
+    angle_threshold=0.05,
+    angular_velocity=1.0,
+    distance_threshold=0.25,
+    linear_velocity=1.0,
+    vdb=None,
+    render_debug_video=False,
+):
+    """
+    Return True if the robot can navigate from point A
+    to point B given the configuration of the navmesh.
+    """
+    logger.info(
+        "Checking robot navigability between target object start and goal:"
+    )
+
+    # TODO: for now computation is repeated here, but should be provided from earlier
+    largest_island_id = get_largest_island_index(
+        sim.pathfinder, sim, allow_outdoor=False
+    )
+
+    snapped_start_pos = sim.pathfinder.snap_point(start_pos, largest_island_id)
+    snapped_goal_pos = sim.pathfinder.snap_point(goal_pos, largest_island_id)
+
+    logger.info(
+        f"     - start_pos to snapped_start_pos distance = {(snapped_start_pos-start_pos).length()}"
+    )
+    logger.info(
+        f"     - goal_pos to snapped_goal_pos distance = {(snapped_goal_pos-goal_pos).length()}"
+    )
+
+    if render_debug_video:
+        assert vdb is not None, "Need a vdb for visual debugging."
+        sim.navmesh_visualization = True
+
+    # Create a new pathfinder with slightly stricter radius to provide nav buffer from collision
+    pf = habitat_sim.nav.PathFinder()
+    modified_settings = sim.pathfinder.nav_mesh_settings
+    modified_settings.agent_radius += 0.05
+    assert sim.pathfinder.nav_mesh_settings != modified_settings
+    assert sim.recompute_navmesh(
+        pf, modified_settings
+    ), "failed to recompute navmesh"
+    # Init the shortest path
+    path = habitat_sim.ShortestPath()
+    # Set the locations
+    path.requested_start = snapped_start_pos
+    path.requested_end = snapped_goal_pos
+    # Find the path
+    pf.find_path(path)
+    curr_path_points = path.points
+    if len(curr_path_points) <= 2:
+        return 1.0
+    # Set the initial position
+    p0 = mn.Vector3(curr_path_points[0])
+    p1 = mn.Vector3(curr_path_points[1])
+    p1[1] = curr_path_points[0][1]
+    trans: mn.Matrix4 = mn.Matrix4.look_at(
+        eye=p0, target=p1, up=mn.Vector3(0.0, 1.0, 0.0)
+    )
+
+    # Get the robot position
+    robot_pos = np.array(trans.translation)
+    # Get the navigation target
+    final_nav_targ = np.array(curr_path_points[-1])
+    obj_targ_pos = np.array(curr_path_points[-1])
+    # the velocity control
+    vc = SimpleVelocityControlEnv()
+    forward = np.array([0, 0, -1.0])
+
+    at_goal = False
+
+    collision = []
+    debug_video_frames: List[Any] = []
+    debug_framerate = 30
+    time_since_debug_frame = 9999.0
+
+    try:
```

Comment:
Remove the try catch completely and replace with error handling. This will make it hard to debug or catch problems.